src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,94 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_FACTOR = 2;
+    private int capacity;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        capacity = INITIAL_CAPACITY;
+        table = new Node[INITIAL_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int index = getIndex(key);
+        if (size == (capacity * LOAD_FACTOR)) {
+            resize();
+        }
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+        } else {
+            Node<K, V> nodeByIndex = table[index];
+            while (nodeByIndex != null) {
+                if (Objects.equals(nodeByIndex.key, key)) {
+                    nodeByIndex.value = value;
+                    return;
+                }
+                if (nodeByIndex.next == null) {
+                    nodeByIndex.next = newNode;
+                    size++;
+                    return;
```

Comment:
replace a `return` by a `break` and than remove `size++` from 28th and 38th lines and place it at the very end of the file